### ISSUE TITLE
Poetry fails to install enum 1.1.6 on Python 2.7.5

### ISSUE THREAD

**Issue by newinone** [sdispater/poetry#859]
poetry install -vvv
Virtualenv pason-crypto-py2.7 already exists.
Using virtualenv: /home/yniu/.cache/pypoetry/virtualenvs/pason-crypto-py2.7
Installing dependencies from lock file
Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.

[SolverProblemError]
Because pason-crypto depends on enum34 (1.1.6) which doesn't match any versions, version solving failed.

Exception trace:
 /home/yniu/.poetry/lib/poetry/_vendor/py2.7/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /home/yniu/.poetry/lib/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /home/yniu/.poetry/lib/poetry/_vendor/py2.7/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /home/yniu/.poetry/lib/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /home/yniu/.poetry/lib/poetry/_vendor/py2.7/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /home/yniu/.poetry/lib/poetry/_vendor/py2.7/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /home/yniu/.poetry/lib/poetry/console/commands/install.py in handle() at line 57
   return_code = installer.run()
 /home/yniu/.poetry/lib/poetry/installation/installer.py in run() at line 76
   self._do_install(local_repo)
 /home/yniu/.poetry/lib/poetry/installation/installer.py in _do_install() at line 216
   ops = solver.solve(use_latest=whitelist)
 /home/yniu/.poetry/lib/poetry/puzzle/solver.py in solve() at line 38
   packages, depths = self._solve(use_latest=use_latest)
 /home/yniu/.poetry/lib/poetry/puzzle/solver.py in _solve() at line 180
   raise SolverProblemError(e)

- [X] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

Was able to install by pip
 sudo pip install enum34
Collecting enum34
  Downloading https://files.pythonhosted.org/packages/af/42/cb9355df32c69b553e72a2e28daee25d1611d2c0d9c272aa1d34204205b2/enum34-1.1.6-py3-none-any.whl
Installing collected packages: enum34
Successfully installed enum34-1.1.6

- **OS version and name**: 
- **Poetry version**: 
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**:

**Comment by bertjwregeer** [comment]
Using pip you are downloading enum34 using Python 3, however your poetry environment looks to be Python 2.7.

```
Downloading https://files.pythonhosted.org/packages/af/42/cb9355df32c69b553e72a2e28daee25d1611d2c0d9c272aa1d34204205b2/enum34-1.1.6-py3-none-any.whl
```

Is the Python 3 version.

Your lock file is also out of date with your `pyproject.toml` file.

You also did not link to a gist with your `pyproject.toml` file.

I am unable to replicate your issue:

# `pyproject.toml`
```
[tool.poetry]
name = "testing"
version = "0.1.0"
description = ""
authors = ["Your Name <you@example.com>"]

[tool.poetry.dependencies]
python = "*"
enum34 = "^1.1"

[tool.poetry.dev-dependencies]
pytest = "^3.5"
```

```
Riley:testing xistence$ poetry install -vvv
Using virtualenv: /Users/xistence/Projects/testing/.venv
Installing dependencies from lock file


Package operations: 11 installs, 0 updates, 0 removals, 1 skipped

  - Installing scandir (1.9.0)
  - Installing six (1.12.0)
  - Installing atomicwrites (1.3.0)
  - Installing attrs (18.2.0)
  - Skipping colorama (0.4.1) Not needed for the current environment
  - Installing funcsigs (1.0.2)
  - Installing more-itertools (5.0.0)
  - Installing pathlib2 (2.3.3)
  - Installing pluggy (0.8.1)
  - Installing py (1.7.0)
  - Installing enum34 (1.1.6)
  - Installing pytest (3.10.1)
  - Installing testing (0.1.0)
Riley:testing xistence$ poetry run python -V
Python 2.7.5
```
